### ISSUE TITLE
UseLet is reported for else if

### ISSUE THREAD

**Issue by matejdro** [detekt/detekt#7134]
## Expected Behavior

Following code should not trigger `UseLet` rule:

```kotlin
val a: Int? = null
val b = if (true) { // Any condition can go here
   0 // Anything can go here
} else if (a != null) {
   a + 1
} else {
   null
}
```

## Observed Behavior

Above code triggers `UseLet` rule on the `else if` line

## Steps to Reproduce

1. Attempt to scan above code with detekt

## Context

While it is true, that above code could technically be turned into let, maybe something like that:

```kotlin
val c = if (true) {
   0
} else {
   a?.let { it + 1}
}
```

I feel like this variant is less readable and inferior to the else if variant.

## Your Environment

* Version of detekt used: 1.23.6
* Version of Gradle used (if applicable): 8.7

**Comment by BraisGabin** [comment]
I'm not sure about this. I don't see why the else if case should be treated differently. Maybe on a more real use case it is more clear, but in your snippet I think that the `let` works prefect.

**Comment by matejdro** [comment]
Because conversion from else if to let is much more messy than from if to let:

* in if -> let, let replaces the entire let statement, so the complexity is about the same than before, it's just clearer to the user (because ?.let usually means null check)
* in if else -> let, let does not replace the if statement. Instead it adds another *nested* statement to it. Which in my opinion makes it less readable and more complex (I think it's even  doubly more complex by the detekt's own cyclomatic complexity rule, because it's nested)

**Comment by BraisGabin** [comment]
I would say not to add that on detekt but I would like to know what other @detekt/maintainers think about it.

**Comment by schalkms** [comment]
I already voted for **not add**.

**Comment by BraisGabin** [comment]
Then closing for the reasons explained before. In any case, thanks for the proposal :)
